File: packages/sdk-python/veltrix_sdk/resources/brand.py
# -*- coding: utf-8 -*-

from .base import BaseResource


class BrandResource(BaseResource):
    """Public branding endpoint (``/api/brand``).

    Returns the deployment's branding (name, tagline, logo URL). Public — no
    authentication required.
    """
    RESOURCE_PATH = "brand"

    def get(self, **kwargs):
        """Retrieves the deployment branding (GET /api/brand)."""
        return self._list(**kwargs)

File: packages/sdk-js/src/resources/brand.ts
import { BaseResource } from './base';
import { AxiosRequestConfig } from 'axios';

// Public branding endpoint (/api/brand). Returns the deployment's branding
// (name, tagline, logo URL). Public — no authentication required.

interface Brand {
  name: string;
  tagline?: string | null;
  logoUrl?: string | null;
}

export class BrandResource extends BaseResource {
  protected readonly RESOURCE_PATH = 'brand';

  /** Retrieves the deployment branding. GET /api/brand */
  async get(config?: AxiosRequestConfig): Promise<Brand> {
    return this._list<Brand>(undefined, config);
  }
}
